src/PJ_eqearth.c

```diff
@@ -28,28 +30,59 @@ PROJ_HEAD(eqearth, "Equal Earth") "\n\tPCyl., Sph.";
 #define EPS 1e-11
 #define MAX_ITER 12
 
-static XY s_forward (LP lp, PJ *P) {           /* Spheroidal, forward */
+struct pj_opaque {
+    double  qp;
+    double  rqda;
+    double *apa;
+};
+
+static XY e_forward (LP lp, PJ *P) {           /* Ellipsoidal/Spheroidal, forward */
     XY xy = {0.0,0.0};
-    double phi, phi2, phi6;
-    (void) P;
+    struct pj_opaque *Q = P->opaque;
+    double sbeta;
+    double psi, psi2, psi6;
+
+    if (P->es != 0.0) {
+       /* Ellipsoidal case, converting to authalic latitude */
+       sbeta = pj_qsfn(sin(lp.phi), P->e, 1.0 - P->es) / Q->qp;
+       if (fabs(sbeta) > 1) {
+          /* Rounding error. */
+          sbeta = sbeta > 0 ? 1 : -1;
```

Comment:
It would be nice to have a test case that exercises this condition.

Reply:
I am not sure I can hit this test case. The best testing case is probably phi = +/-90. In this case pj_qsfn() function returns +/-Q->qp and for some rounding error it might hit this case. Those tests are already in and looked they are passing.

Reply:
Using phi > 90 would not help, because sin function would already "fix" this.

Reply:
All right, we'll live without a test for that